src/org/recompile/mobile/PlatformPlayer.java

```diff
@@ -208,6 +208,8 @@ private class midiPlayer extends audioplayer
 
 		private int loops = 0;
 
+		private Long tick = new Long(0);

```

Comment:
Believe it or not, it's actually better to prefer the primitive to the reference type most of the time.   [This site makes an okay case](https://medium.com/geekculture/effective-java-prefer-primitive-types-to-boxed-types-ea02eb9b8c62) 

It's a little annoying with Longs in java, but there's a suffix you can use to avoid a cast.
`private Long tick = 0L;`

Reply:
I see... yes, makes sense to use primitives there. wavPlayer also has an assignment like this one for its "private Long time" instance variable, so i'll change it to a primitive as well.